src/mlpack/core/tree/cosine_tree/cosine_tree.cpp

```diff
@@ -150,6 +150,209 @@ CosineTree::CosineTree(const arma::mat& dataset,
   ConstructBasis(treeQueue);
 }
 
+//! Copy the given tree.
+CosineTree::CosineTree(const CosineTree& other) :
+  //dataset((other.parent == NULL) ? other.dataset : NULL)
```

Comment:
For this. You just do `dataset(other.dataset)` temporarily in the `initializer list` (to remove the initialization error) and then do `deep copy` as mentioned in my previous comments.

Reply:
https://github.com/mlpack/mlpack/pull/2023#discussion_r331147450